pyccel/codegen/printing/cwrappercode.py

```diff
@@ -632,6 +638,9 @@ def _print_Interface(self, expr):
             # Building PybuildValue and freeing the allocated variable after.
             mini_wrapper_func_body.append(AliasAssign(wrapper_results[0],PyBuildValueNode(res_args)))
             mini_wrapper_func_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+            # Call free function for C type
+            if self._target_language == 'c':
+                mini_wrapper_func_body += [Deallocate(i) for i in func.arguments if i.allocatable]
```

Comment:
does the function arguments have the attr `allocatable` == 'true'?

shouldn't it be set as `false` and let the argument just be a pointer?

Reply:
any variable with rank > 0 is set as allocatable, so checking for rank > 0 or allocatable == true for detecting array is the same thing in the c wrapper, look at the deallocator print inside `cwrappercode ` it is different then ccode deallocator print.